### ISSUE TITLE
Errors when running 3D Angular app in production mode

### ISSUE THREAD

**Issue by sullivani** [Esri/jsapi-resources#346]
**Describe the bug**
I am getting the following errors when running a simple 3D Angular application in production mode. When the app is run in development mode, no such errors occur. These errors do not occur in an equivalent 2D Angular app. 

As a result of these errors, the scene layers do not display at all.

![image](https://user-images.githubusercontent.com/51807829/136868547-8a9a1cee-21de-4278-b6f8-23b92615d179.png)

**Steps to Reproduce**
Clone and run the example application [here ](https://bitbucket.org/mirvac-research/jsapi-scene-issue/src/master/) to reproduce.

You will see that the Angular app produces the above errors in the browser.

**Comment by sullivani** [comment]
Hi @andygup looks like I spoke to soon....I am running into additional issues. Its a similar issue in that it only occurs in production mode.

I can see now the scene layer, but the browser throws up the following error. I also noticed that when you pan/zoom the map, the styling of the scene layer is not applied correctly i.e. the building floor outline disappears (as per screenshot). 

![image](https://user-images.githubusercontent.com/51807829/136876142-ea068762-111d-4e03-8dd4-d70fed795e29.png)

![image](https://user-images.githubusercontent.com/51807829/136876243-b501971a-ee95-4cb0-9212-20dae6cab692.png)

This issue can be replicated using the example application [here ](https://bitbucket.org/mirvac-research/jsapi-scene-issue/src/master/) (I have updated the Angular version).

**Comment by sullivani** [comment]
@andygup No, I have always had issues trying to use scene layers in an Angular app with the JSAPI.
But, in saying that, I have only been working with scene layers more recently i.e. a few months.

**Comment by sullivani** [comment]
Hi @andygup has there been any progress on the styling issue?

**Comment by andygup** [comment]
Hi @sullivani we have not been able to reproduce the building floor outlines disappearing issue. Can you try testing your SceneLayer against our Angular sample? 

https://github.com/Esri/jsapi-resources/tree/master/esm-samples/jsapi-angular-cli

Here is a screen shot of our sample using your SceneLayer in a production build:

![image](https://user-images.githubusercontent.com/510440/140827058-1aa37af1-cf97-449d-82a4-bfd98b56d084.png)

**Comment by sullivani** [comment]
Hi @andygup I was able to run the example with my scene layer and the styling issues were not present.
The only difference between your example app and my custom app is the Angular version, so I assume that Angular is the cause here.

I haven't been able to test the latest version of Angular with my custom app as yet because it is quite complex, so it may take some time to do the upgrade to Angular 13.

**Comment by andygup** [comment]
Could be the Angular version. Something else that you could easily check is the JS API CSS -  I noticed your bitbucket repo didn't have anything in `app,component.css`. Maybe that's something to double check and also compare against our sample.

**Comment by sullivani** [comment]
The JSAPI CSS import was included in the style.css file, that's why it wasn't in the app.component.css.

In my custom app, I am using a custom version of the JSAPI CSS, but to test whether that was causing the issue, I disabled this custom CSS and imported the default JSAPI CSS and this still resulted in the same issues.

I am fairly certain now its the Angular version, but wont know for sure until I complete the upgrade.

**Comment by sullivani** [comment]
Here is some more info regarding this styling issue.

I downgraded the Angular app [https://github.com/Esri/jsapi-resources/tree/master/esm-samples/jsapi-angular-cli](https://github.com/Esri/jsapi-resources/tree/master/esm-samples/jsapi-angular-cli) to v12.1.4 and ran it in production mode, and I was able to replicate the issue.

I then upgraded the Angular app back to v13, ran it in production mode and the issue is not present.

Based on that, I am almost 100% certain its the Angular version.
